extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/KinesisRecordSupplier.java

```diff
@@ -296,14 +279,22 @@ private Runnable fetchRecords()
             );
 
 
-            log.trace(
-                "Stream[%s] / partition[%s] / sequenceNum[%s] / bufferRemainingCapacity[%d]: %s",
-                currRecord.getStream(),
-                currRecord.getPartitionId(),
-                currRecord.getSequenceNumber(),
-                records.remainingCapacity(),
-                currRecord.getData().stream().map(StringUtils::fromUtf8).collect(Collectors.toList())
-            );
+            if (log.isTraceEnabled()) {
```

Comment:
I added this check, to avoid unnecessarily deserializing all the bytes as strings

Reply:
Nice 😄

I bet it will speed up Kinesis ingestion a bit.

Reply:
let's hope so, using ByteEntities directly should also avoid having to copy around byte arrays